src/bootstrap.coffee

```diff
@@ -150,15 +154,20 @@ bootstrapper.done = new Promise (resolve) ->
 		resolve(userConfig)
 		# If we're still using an old api key we can try to exchange it for a valid device key
 		if userConfig.apiKey?
-			exchangeKey()
-			.then ->
-				delete userConfig.apiKey
-				knex('config').update(value: userConfig.deviceApiKey).where(key: 'apiKey')
-				.then ->
-					fs.writeFileAsync(configPath, JSON.stringify(userConfig))
+			Promise.join(
+				osRelease.getOSVersion(config.hostOSVersionPath)
+				exchangeKey()
+				(osVersion) ->
+					# Only delete the provisioning key if we're on a Resin OS version that supports using the deviceApiKey (2.0.2 and above)
+					# Otherwise VPN and other host services that use an API key will break.
+					hasDeviceApiKeySupport = /^Resin OS /.test(osVersion) and semver.gte(semverRegex.test(osVersion), '2.0.2')
```

Comment:
I think we would want to delete the key for custom OS builds as well, since they should be based on an up to date enough meta-resin to work (and we don't want to have the case that custom os builds keep the provisioning key around forever)